internal/host/host.go

```diff
@@ -1138,3 +1148,25 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 func (m *Manager) UnRegisterHost(ctx context.Context, hostID, clusterID string) error {
 	return m.db.Where("id = ? and cluster_id = ?", hostID, clusterID).Delete(&common.Host{}).Error
 }
+
+func (m *Manager) captureConnectivityReportMetrics(ctx context.Context, openshiftVersion string, h *models.Host, report string, hosts []*models.Host) {
+	log := logutil.FromContext(ctx, logrus.New())
+	defer commonutils.MeasureOperation("CaptureConnectivityReportMetrics", log, m.metricApi)
+
+	connectivityReport, err := hostutil.UnmarshalConnectivityReport(h.Connectivity)
+	if err != nil {
+		log.Errorf("unable to unmarshal connectivity report for host ID %s:%v", h.ID, err)
+		return
+	}
+	for _, r := range connectivityReport.RemoteHosts {
+		for _, l3 := range r.L3Connectivity {
+			_, targetRole, err := GetHostnameAndRoleByIP(l3.RemoteIPAddress, hosts)
+			if err != nil {
+				log.Warn(err)
+				continue
+			}
+			m.metricApi.NetworkLatencyBetweenHosts(openshiftVersion, h.Role, targetRole, l3.AverageRTTMs)
```

Comment:
Seems strange the host is not part of the metrics.  If there was a problem - don't we want to know which hosts were involved ?

Reply:
Adding the host as a label would literally explode the metrics' cardinality.